### PR TITLE
[IOTDB-1450] Deletion should only delete related time partitions

### PR DESCRIPTION
**Pull request by jt2594838**
https://issues.apache.org/jira/browse/IOTDB-1450

### CODE REVIEW

**Comment by jt2594838** [comment]
The failing tests are caused by spotless, but strangely when I use `mvn spotless:apply`, nothing changes.

File: cluster/src/main/java/org/apache/iotdb/cluster/log/applier/DataLogApplier.java [hunk 1]
```diff
@@ -79,6 +81,12 @@ public void apply(Log log) {
       } else if (log instanceof PhysicalPlanLog) {
         PhysicalPlanLog physicalPlanLog = (PhysicalPlanLog) log;
         PhysicalPlan plan = physicalPlanLog.getPlan();
+        if (plan instanceof DeletePlan) {
```
[the remark this answers is not in the document]
Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/applier/DataLogApplier.java:84]: Yes, I was just thinking whether we need to add this logic when replicaNum == 1 for `PlanExecutor` when `TimePartition` is enabled.
 Although  we don't have raft logic in this mode, but different timePartitions in same storage group can be managed by multiple dataGroups on a node.
 What's your opinion?

[the remark this answers is not in the document]
Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/applier/DataLogApplier.java:84]: Yes, I was just thinking whether we need to add this logic when replicaNum == 1 for PlanExecutor when TimePartition is enabled.
Although we don't have raft logic in this mode, but different timePartitions in same storage group can be managed by multiple dataGroups on a node.
What's your opinion?

[the remark this answers is not in the document]
Comment (neuyilan) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/applier/DataLogApplier.java:84]: Yes, I think so, we can extract a method in the PlanExecutor to execute these logics when the replicaNum ==1.

[the remark this answers is not in the document]
Comment (neuyilan) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/applier/DataLogApplier.java:84]: I think we can merge the PR first and resolve the above issue and some sonar failed in the next PR.

**Comment by HTHou** [comment]
The conflict is fixed.

File: cluster/src/main/java/org/apache/iotdb/cluster/log/applier/DataLogApplier.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/applier/DataLogApplier.java:84]: I have left my review in this [PR](https://github.com/apache/iotdb/pull/3879), maybe we can add this issue in it.
PTAL.

**Pull request merged**